x/projects/keeper/project_test.go

```diff
@@ -357,3 +436,126 @@ func SetPolicyTest(t *testing.T, testAdminPolicy bool) {
 		})
 	}
 }
+
+func TestAddDevKeyToSameProjectDifferentBlocks(t *testing.T) {
+	_, keepers, ctx := testkeeper.InitAllKeepers(t)
+
+	projectName := "mockname1"
+	subAccount := common.CreateNewAccount(ctx, *keepers, 10000)
+	devAcc1 := common.CreateNewAccount(ctx, *keepers, 10000)
+	devAcc2 := common.CreateNewAccount(ctx, *keepers, 10000)
+	projectID := types.ProjectIndex(subAccount.Addr.String(), projectName)
+	plan := common.CreateMockPlan()
+
+	projectData := types.ProjectData{
+		Name:        projectName,
+		Description: "",
+		Enabled:     true,
+		ProjectKeys: []types.ProjectKey{{
+			Key:   subAccount.Addr.String(),
+			Types: []types.ProjectKey_KEY_TYPE{types.ProjectKey_DEVELOPER},
+			Vrfpk: "",
+		}},
+		Policy: &plan.PlanPolicy,
+	}
+	err := keepers.Projects.CreateProject(sdk.UnwrapSDKContext(ctx), subAccount.Addr.String(), projectData, plan)
+	require.Nil(t, err)
+
+	ctx = testkeeper.AdvanceBlock(ctx, keepers)
+
+	err = keepers.Projects.AddKeysToProject(sdk.UnwrapSDKContext(ctx), projectID, subAccount.Addr.String(),
+		[]types.ProjectKey{{
+			Key:   devAcc1.Addr.String(),
+			Types: []types.ProjectKey_KEY_TYPE{types.ProjectKey_DEVELOPER},
+			Vrfpk: "",
+		}})
+	require.Nil(t, err)
+
+	ctx = testkeeper.AdvanceBlock(ctx, keepers)
+
+	err = keepers.Projects.AddKeysToProject(sdk.UnwrapSDKContext(ctx), projectID, subAccount.Addr.String(),
+		[]types.ProjectKey{{
+			Key:   devAcc2.Addr.String(),
+			Types: []types.ProjectKey_KEY_TYPE{types.ProjectKey_DEVELOPER},
+			Vrfpk: "",
+		}})
+	require.Nil(t, err)
+
+	proj, _, err := keepers.Projects.GetProjectForDeveloper(sdk.UnwrapSDKContext(ctx), subAccount.Addr.String(),
+		uint64(sdk.UnwrapSDKContext(ctx).BlockHeight()))
+	require.Nil(t, err)
+
+	require.Equal(t, 3, len(proj.ProjectKeys))
+}
+
+func TestAddDevKeyToDifferentProjectsInSameBlock(t *testing.T) {
+	_, keepers, ctx := testkeeper.InitAllKeepers(t)
+	plan := common.CreateMockPlan()
+
+	subAccount1 := common.CreateNewAccount(ctx, *keepers, 10000)
+	subAccount2 := common.CreateNewAccount(ctx, *keepers, 10000)
+	devAcc := common.CreateNewAccount(ctx, *keepers, 10000)
+
+	projectName1 := "mockname1"
+	projectName2 := "mockname2"
+
+	projectID1 := types.ProjectIndex(subAccount1.Addr.String(), projectName1)
+	projectID2 := types.ProjectIndex(subAccount2.Addr.String(), projectName2)
+
+	projectData1 := types.ProjectData{
```

Comment:
can you reuse or enhance the existing `prepareProjectsData()`?

Reply:
there is no such function?

Reply:
it was merged on May 1, and may bring merge-conflict mayhem to you.
you should use/enhance it for everywhere you prepare a "types.ProjectData{ ... }"